src/parser/ast/FunctionNode.h

```diff
@@ -40,6 +40,36 @@ class FunctionNode : public StatementNode {
     virtual ASTNodeType type() override { return ASTNodeType::Function; }
     virtual void generateStatementByteCode(ByteCodeBlock* codeBlock, ByteCodeGenerateContext* context) override
     {
+        // binding function name if needs
+        if (context->m_codeBlock->functionName().string()->length()) {
+            if (context->m_codeBlock->isFunctionNameSaveOnHeap() || context->m_codeBlock->isFunctionNameUsedBySelf()) {
+                codeBlock->pushCode(BindingCalleeIntoRegister(ByteCodeLOC(m_loc.index)), context, this);
+            }
+        }
+
+        // init stack-allocated vars
+        if (context->m_codeBlock->canUseIndexedVariableStorage()) {
+            auto varInfo = codeBlock->m_codeBlock->identifierInfos();
+            auto fnName = codeBlock->m_codeBlock->functionName();
+            for (size_t i = 0; i < varInfo.size(); i++) {
+                const auto& var = varInfo[i];
+                if (var.m_name == fnName && !var.m_isExplicitlyDeclaredOrParameterName) {
+                    // this case, init from outside
+                    continue;
+                }
+                if (var.m_needToAllocateOnStack && !var.m_isParameterName) {
+                    codeBlock->pushCode(LoadLiteral(ByteCodeLOC(m_loc.index), REGULAR_REGISTER_LIMIT + var.m_indexForIndexedStorage, Value()), context, this);
+                }
+            }
+        }
+
+        // init literal values
+        if (context->m_numeralLiteralData) {
+            for (size_t i = 0; i < context->m_numeralLiteralData->size(); i++) {
+                codeBlock->pushCode(LoadLiteral(ByteCodeLOC(m_loc.index), REGULAR_REGISTER_LIMIT + VARIABLE_LIMIT + i, context->m_numeralLiteralData->data()[i]), context, this);
+            }
```

Comment:
I wonder that stack initialized by `LoadLiteral` bytecode is faster than initialization through existing loop in the call process.

Reply:
When I move initialization into interpreter, I got a few higher score.
IMO If there is no number literal, the for loop is use less
and
We don't need to load number literal data from isolated memory(not same location with bytecode)
I think saving literal data in bytecode can reducing cpu data cache miss count

Reply:
Sounds great!